tests/tests/test_device.py

```diff
@@ -298,6 +298,38 @@ def _test_delete_authset_OK(self, management_api, devices, **kwargs):
         found = management_api.find_device_by_identity(d.identity, **kwargs)
         assert found
 
+    def _test_delete_authset_preauth_OK(self, management_api, devices, **kwargs):
+        # preauthorize a device
+        aid = 'aid-preauth'
+        device_id = 'id-preauth'
+        iddata = json.dumps({'mac': 'mac-preauth'})
+        key = 'key-preauth'
+
+        req = management_api.make_preauth_req(aid, device_id, iddata, key)
+        _, rsp = management_api.preauthorize(req, **kwargs)
+
+        assert rsp.status_code == 201
+
+        # check device created
+        dev = management_api.get_device(id=device_id, **kwargs)
+        assert dev
+        assert len(dev.auth_sets) == 1
+        assert dev.auth_sets[0].status == 'preauthorized'
+        assert dev.auth_sets[0].id == aid
+
+        # delete auth set, check device deleted
+        rsp = management_api.delete_authset(dev.id, aid, **kwargs)
+        assert rsp.status_code == 204
+
+        try:
+            found = management_api.get_device(id=device_id, **kwargs)
+        except bravado.exception.HTTPError as e:
+            assert e.response.status_code == 404
+
+        # check all other devices intact
+        after_devs = management_api.list_devices(**kwargs)
+        assert len(after_devs) == len(devices)
```

Comment:
it only checks if one of the devices was deleted (if there will be no 404 error on delete and different device has been deleted, test will pass)